frontend/src/components/newsItem/NewsItem.tsx

```diff
@@ -0,0 +1,42 @@
+import { Link } from "react-router-dom";
+import { Typography, ListItem, Divider } from "@mui/material";
+
+import styled from "@mui/styled-engine-sc";
+import getTime from "../../utils/timeModify";
+import { useAppSelector } from "../../store/hooksTyped";
+
+const StyledLink = styled(Link)`
+  color: inherit;
+  text-decoration: none;
+  &:hover {
+    text-decoration: underline;
+  }
+`;
+interface NewsIDProp {
+  id: number;
+}
+
+const NewsItem = ({ id }: NewsIDProp) => {
+  const newsItem = useAppSelector((state) =>
+    state.newsIds.newsItems?.find((item) => item.id == id)
+  );
+
+  const { title, score, by, time, url } = newsItem ?? {};
+
+  const resultTime = getTime(time!);
+
+  return url ? (
```

Comment:
лучше эту проверку вынести наверх, просто будет на 2 отступа меньше снизу)
```suggestion
  if(!url) {
    return null;
  }
  
  return <>
```

Reply:
исправлено